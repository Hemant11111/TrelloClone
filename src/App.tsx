import React from 'react';

import './App.css';
import 'font-awesome/css/font-awesome.min.css';
import HeaderComponent from "./components/header/Header";
import ListCardComponent from "./components/cards/list-cards/ListCardComponent";

function App() {


    return (
        <div className="App">
            <HeaderComponent/>

            <div className="app-container">
                <ListCardComponent/>
            </div>

        </div>
    );
}

export default App;
