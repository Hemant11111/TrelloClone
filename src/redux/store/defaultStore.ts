function getDefaultCards() {
    return {
        "default-card": {
            id: "default-card",
            title: "Default card",
            tasks: [],
            taskIds: [],
            deleted: false,
            updated: false,
            deleting: false,
            updating: false,
            selected: false
        }
    }
}

function getDefaultTasks() {
    return {};
}

export function getDefaultState() {
    return {
        cardReducer: {cards: getDefaultCards()},
        taskReducer: {tasks: getDefaultTasks()}
    }
}