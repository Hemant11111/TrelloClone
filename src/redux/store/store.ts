import { applyMiddleware, createStore } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import rootReducer from "../reducers";

function configureStore(preloadedState: any) {
    const middlewares = [thunk];
    const middlewareEnhancer = applyMiddleware(...middlewares);

    const storeEnhancers = [middlewareEnhancer];

    const composedEnhancer = composeWithDevTools(...storeEnhancers);

    return createStore(
        rootReducer,
        preloadedState,
        composedEnhancer
    );
}

const store = configureStore({});

export default store;