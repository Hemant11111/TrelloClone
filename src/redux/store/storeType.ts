import CardModel from "../../model/Card";
import TaskModel from "../../model/Task";


export interface STORE_TYPE {
    cards: { [key: string]: CardModel },
    tasks: { [key: string]: TaskModel }
}


export interface ROOT_STORE_TYPE {
    cardReducer: CARDS_STATE_TYPE,
    taskReducer: TASKS_STATE_TYPE
}

export interface CARDS_STATE_TYPE {
    cards: { [key: string]: CardModel }
}

export interface TASKS_STATE_TYPE {
    tasks: { [key: string]: TaskModel }
}