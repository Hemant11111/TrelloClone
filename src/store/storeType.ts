import CardModel from "../model/Card";
import TaskModel from "../model/Task";


export interface STORE_TYPE {
    cards: { [key: string]: CardModel },
    tasks: { [key: string]: TaskModel }
}


export interface ROOT_STORE_TYPE {
    cardReducer: { cards: { [key: string]: CardModel } },
    taskReducer: { tasks: { [key: string]: TaskModel } }
}
