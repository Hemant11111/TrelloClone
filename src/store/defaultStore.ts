import { STORE_TYPE } from "./storeType";

export const DEFAULT_STORE: STORE_TYPE = {
    cards: {},
    tasks: {}
}
