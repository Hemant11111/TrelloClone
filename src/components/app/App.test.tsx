import React from 'react';
import { Provider } from "react-redux";
import { render, screen } from '@testing-library/react';

import App from './App';
import store from "../../redux/store/store";


describe('Rendering App component', () => {
    it('should render app component wrapped in provider', () => {
        const {asFragment} = render(
            <Provider store={store}>
                <App/>
            </Provider>
        )
        expect(asFragment()).toMatchSnapshot();
        expect(screen.getByText('Trello-Clone')).toBeInTheDocument();
    })
})