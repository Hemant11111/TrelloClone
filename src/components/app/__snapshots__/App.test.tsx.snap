// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering App component should render app component wrapped in provider 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <h2>
        Trello-Clone
      </h2>
      <button>
        Sign out
      </button>
    </div>
    <div
      class="app-container"
    >
      <div
        class="card card-container"
      >
        <span
          class="card-title"
        >
          Default card
        </span>
        <div
          class="card-tasks-container"
        >
          <div
            class="task-list-container"
          >
            <div
              class="add-task-container"
            >
              <form
                class="custom-form"
              >
                <input
                  class=" input-form"
                  placeholder="Add Task..."
                  value=""
                />
              </form>
              <button
                class="add-task-button trello-icon-button"
                data-testid="add-task-button"
                disabled=""
              >
                <i
                  class="fa fa-plus"
                />
              </button>
            </div>
          </div>
        </div>
        <button
          class="card-dlt-button"
          data-testid="btn-card-delete"
        >
          <i
            class="fa fa-remove"
          />
           Delete
        </button>
      </div>
      <div
        class="add-new-card-container"
      >
        <button
          class="add-new-card-btn"
          data-testid="add-new-card-button"
        >
          <i
            class="fa fa-plus"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
